### PR TITLE
Add more detail to recovery bandwidth limit docs

### PR DESCRIPTION
**Pull request by DaveCTurner**

### CODE REVIEW

File: docs/reference/modules/indices/recovery.asciidoc [hunk 1]
```diff
@@ -1,10 +1,22 @@
 [[recovery]]
 === Indices Recovery
 
-The following _expert_ setting can be set to manage the recovery policy.
+_Recovery_ is the process used to build a new copy of a shard on a node by
+copying data from the primary. {es} uses this recovery process to rebuild shard
+copies that were lost if a node has failed, and uses the same process when
+migrating a shard copy between nodes to rebalance the cluster or to honour any
```
Comment (original-brownbear) [inline docs/reference/modules/indices/recovery.asciidoc]: NIT: I'm sorry :) but maybe `honor` (we're consistently using :us: in our docs as far as I can see)?

File: docs/reference/modules/indices/recovery.asciidoc [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+changes to the <<modules-cluster,shard allocation settings>>.
+
+The following _expert_ setting can be set to manage the resources consumed by
+recoveries:
 
 `indices.recovery.max_bytes_per_sec`::
-    Defaults to `40mb`.
+    Limits the total inbound and outbound recovery traffic on each node. If you
+    set this limit too high then there is a risk that ongoing recoveries will
+    consume an excess of bandwidth (or other resources) which could destabilise
```
Comment (original-brownbear) [inline docs/reference/modules/indices/recovery.asciidoc]: NIT: `destabilize` ?

**Review by original-brownbear — APPROVED** [review]
LGTM

File: docs/reference/modules/indices/recovery.asciidoc [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+recoveries:
 
 `indices.recovery.max_bytes_per_sec`::
-    Defaults to `40mb`.
+    Limits the total inbound and outbound recovery traffic on each node. If you
+    set this limit too high then there is a risk that ongoing recoveries will
+    consume an excess of bandwidth (or other resources) which could destabilize
+    the cluster. Since this setting applies to each node separately, the total
+    bandwidth that might be used by recoveries is the value of this setting
+    multiplied by half of the number of nodes in the cluster. Defaults to
```
Comment (ywelsch) [inline docs/reference/modules/indices/recovery.asciidoc]: I don't understand this sentence and the formula, and find it rather more confusing than helpful. Maybe omit this?

File: docs/reference/modules/indices/recovery.asciidoc [hunk 4]
```diff
@@ -1,10 +1,23 @@
 [[recovery]]
 === Indices Recovery
 
-The following _expert_ setting can be set to manage the recovery policy.
+_Recovery_ is the process used to build a new copy of a shard on a node by
```
Comment (ywelsch) [inline docs/reference/modules/indices/recovery.asciidoc]: you're giving a definition of peer recoveries here, not recoveries in general (see e.g. docs for `_cat/recovery`)

Comment (ywelsch) [reply docs/reference/modules/indices/recovery.asciidoc]: perhaps we can crosslink to those docs for the explanation of what recoveries are

**Review by ywelsch — CHANGES_REQUESTED** [review]
Thanks for improving the docs. I've left some comments.

File: docs/reference/modules/indices/recovery.asciidoc (hunk 3 above)
Comment (DaveCTurner) [reply docs/reference/modules/indices/recovery.asciidoc]: I would like somehow to indicate that although `40MB/s` seems like a small number, it's multiplied by the number of nodes, so 8 nodes all recovering at their maximum speed will saturate a 1Gbit network with recoveries. This seems to be a point of confusion.

Comment (ywelsch) [reply docs/reference/modules/indices/recovery.asciidoc]: I'm not sure I understand. The rate limiter works both at the sender and receiver side of file chunks to ensure that even for all recoveries combined we only use `40MB/s` on the connections to/from a single node. Depending on the network topology (e.g. not all nodes connected to the same switch), other bandwidth restrictions might exist. I think it's ok to point out that this limit is a per-node thing and does not account for the overall recovery traffic in the cluster, but I would avoid adding this formula which feels more like a "Milchmädchenrechnung" to me.

Comment (DaveCTurner) [reply docs/reference/modules/indices/recovery.asciidoc]: Ok, weakened in d4ab7a8

**Review by ywelsch — APPROVED** [review]
Thanks

**Pull request merged**